### ISSUE TITLE
windows: 1.4.0-rc.4 -- RtAudio hangs

### ISSUE THREAD

**Issue by cchafe** [jacktrip/jacktrip#408]
On Win10 release 1.4.0-rc.4 doesn't display the GUI but the previous release does. 
I cycled between them a couple times with uninstall just to be sure. 
Double-clicking the icon puts it in background and running the .exe to launch the GUI from powershell -- same thing.
Running the cli version is ok.

**Comment by wholenotes** [comment]
Yes, same results here--no GUI but command line works fine. Aaron and I have also been messaging about the fact that QJackTrip needs to be manually removed before 1.4 installer will proceed.

Bonnie

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by psiborg112** [comment]
Would someone who's having trouble with the GUI be able to try this 
build: https://www.psi-borg.org/Test.msi

It's 1.4.0-rc4 but built with CMake. This will point us in the direction 
of whether it's a code or a build issue. (It runs on my machine, but it 
would be good for that to be confirmed elsewhere.)

Cheers,
Aaron

**Comment by wholenotes** [comment]
GUI comes up now. Received connection from Peer! Thanks!

Bonnie

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by dyfer** [comment]
TL;DR This seems to be related to the RtAudio library which we get from VCPKG and specifically to the ASIO backend. It seems that RtAudio has been [updated](https://github.com/microsoft/vcpkg/pull/19583) before our rc.4.

(I was quite puzzled since not much changed on the build front between rc.3 and rc.4)

To check that this is indeed related to RtAudio: on my system rc.4 also fails in command line if I try to use rtaudio (e.g. `jacktrip --listdevices` which will query devices from rtaudio). Aaron's build uses RtAudio, but built without ASIO backend. As I wrote before, this is likely specific to the VCPKG build of RtAudio and I don't assume that Aaron is using VCPKG anyway (?)

I'm working on a fix, trying to figure out how to force an older version of the RtAudio package in VCPKG

**Comment by psiborg112** [comment]
Yep, that would explain it, and glad you figured it out. (Was definitely 
surprised that it would be crashing like this because not much has 
changed on the code front recently either, and definitely nothing that 
would seem to explain this.) Not using VCPKG and haven't updated RtAudio 
recently.

**Comment by dyfer** [comment]
Incidentally, have you worked much with WASAPI RtAudio on Windows? For now we only recommend ASIO but in the future we could consider other backends, if the performance is adequate (whatever that means).

**Comment by dyfer** [comment]
@wholenotes would you mind testing builds from #409 to make sure the issue is fixed?

**Comment by psiborg112** [comment]
I haven't. Mostly I'm not on Windows, and when I am I'm usually using JACK. I should really look into using the ASIO backend for RtAudio in my builds though.

**Comment by wholenotes** [comment]
GUI shows up for me on Win10 now.

**Comment by ntonnaett** [comment]
This looks like this is actually a bug in RtAudio as #409 only changes RtAudio to an older commit. Maybe it's already fixed in the latest commit?! @dyfer Did you test this?

**Comment by dyfer** [comment]
Good point @ntonnaett , I haven't tested this. I assumed that this is due only to the build config change in VCPKG and not rtaudio itself... But maybe I'm wrong. 

Now thinking about it - the issue I have with integrating rtaudio in qmake (another PR) does have the same result - it compiles fine, but JackTrip freezes when trying to access rtaudio...

**Comment by cchafe** [comment]
Should I remove rc4 so as not to confound folks who were encouraged to use it? I had sent some notices around.
Assuming a release can be removed...

**Comment by ntonnaett** [comment]
Let's merge Marcin's workaround and release rc.5.Am 17.09.2021 17:15 schrieb cchafe ***@***.***>:
Should I remove rc4 so as not to confound folks who were encouraged to use it? I had sent some notices around.
Assuming a release can be removed...

**Comment by dyfer** [comment]
We could take down Windows downloads and leave the rest up, or remove all downloads but leave the release up... or remote the release altogether (but still leave the tag in). All can be done in the GH web ui.

I agree with Nils, the quickest way to make JT work again is to merge my workaround and release rc.5 asap, and then we'll figure out what the "proper" solution is

**Comment by dyfer** [comment]
Based on the test with QMake RtAudio PR it does look like it's an issue with RtAudio itself, not with VCPKG.

**Comment by dyfer** [comment]
I've confirmed it is a bug in RtAudio: https://github.com/thestk/rtaudio/issues/323

**Comment by cchafe** [comment]
I wrote it up on the release page. Thanks!
